packages/toolpad-app/src/toolpad/AppEditor/HierarchyExplorer/CreateConnectionNodeDialog.tsx

```diff
@@ -107,12 +108,24 @@ export default function CreateConnectionDialog({
             onChange={(event) => setDataSourceType(event.target.value)}
           >
             {(Object.entries(dataSources) as ExactEntriesOf<typeof dataSources>).map(
-              ([type, dataSourceDef]) =>
-                dataSourceDef && (
-                  <MenuItem key={type} value={type}>
-                    {dataSourceDef.displayName}
-                  </MenuItem>
-                ),
+              ([type, dataSourceDef]) => {
+                const isDisabledInDemo = config.isDemo && !dataSourceDef?.isDemoFeature;
+                const isSelectable = dataSourceDef && !dataSourceDef.isSingleQuery;
```

Comment:
`isSingleQuery` defines that a datasource should not show up in the connection select because it can only have one type of query (it's not configurable) - in this case the local movies "API"

Reply:
I wonder whether it makes sense to encode this by making the `ConnectionParamsInput` optional? I mean, that's essentially what this does, right?